contracts/peripheral/YieldDirectorGohm.sol

```diff
@@ -0,0 +1,363 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import {IERC20} from "../interfaces/IERC20.sol";
+import {IsOHM} from "../interfaces/IsOHM.sol";
+import {IgOHM} from "../interfaces/IgOHM.sol";
+import {IStaking} from "../interfaces/IStaking.sol";
+import {SafeERC20} from "../libraries/SafeERC20.sol";
+import {YieldSplitter} from "../types/YieldSplitter.sol";
+import {OlympusAccessControlled, IOlympusAuthority} from "../types/OlympusAccessControlled.sol";
+
+/**
+    @title YieldDirector (codename Tyche) 
+    @notice This contract allows donors to deposit their sOHM and donate their rebases
+            to any address. Donors will be able to withdraw their principal
+            sOHM at any time. Donation recipients can also redeem accrued rebases at any time.
+ */
+contract YieldDirectorGohm is YieldSplitter, OlympusAccessControlled {
+    using SafeERC20 for IERC20;
+
+    // drop sOHM for mainnet launch
+    address public immutable sOHM;
+    IStaking public immutable staking;
+
+    bool public depositDisabled;
+    bool public withdrawDisabled;
+    bool public redeemDisabled;
+
+    event Deposited(address indexed donor_, address indexed recipient_, uint256 amount_);
+    event DepositUpdated(address indexed donor_, address indexed recipient_, uint256 amount_);
+    event Withdrawn(address indexed donor_, address indexed recipient_, uint256 amount_);
+    event AllWithdrawn(address indexed donor_, uint256 indexed amount_);
+    event Donated(address indexed donor_, address indexed recipient_, uint256 amount_);
+    event Redeemed(address indexed recipient_, uint256 amount_);
+    event EmergencyShutdown(bool active_);
+
+    constructor (
+    	address sOhm_,
+    	address gOhm_,
+    	address staking_,
+    	address authority_
+    )
+        OlympusAccessControlled(IOlympusAuthority(authority_))
+        YieldSplitter(gOhm_)
+    {
+        require(sOhm_ != address(0), "Invalid address for sOHM");
+        require(gOhm_ != address(0), "Invalid address for gOHM");
+        require(staking_ != address(0), "Invalid address for staking");
+
+        sOHM = sOhm_;
+        staking = IStaking(staking_);
+    }
+
+    /************************
+    * Modifiers
+    ************************/
+    modifier isValidDeposit(uint256 amount_, address recipient_) {
```

Comment:
I think were trying to move away from modifiers. While they provide nice visual benefit, under the hood it seems like are not more efficient than internal functions in reducing code size due to how they're implemented. 

I think internal function returning true or false should do the same job more efficiently.

Heres someone raising that issue up. 
https://github.com/ethereum/solidity/issues/6584

Reply:
Cool, will adjust this